catalogue_graph/src/ingestor/extractors/works_extractor.py

```diff
@@ -0,0 +1,97 @@
+import time
+from collections.abc import Generator
+
+from pydantic import BaseModel
+
+import config
+from ingestor.models.denormalised.work import DenormalisedWork
+from ingestor.models.neptune.query_result import WorkConcept, WorkHierarchy
+from ingestor.queries.work_queries import (
+    WORK_ANCESTORS_QUERY,
+    WORK_CHILDREN_QUERY,
+    WORK_CONCEPTS_QUERY,
+    WORK_QUERY,
+)
+from utils.elasticsearch import get_client, get_standard_index_name
+
+from .base_extractor import GraphBaseExtractor
+
+
+class ExtractedWork(BaseModel):
+    work: DenormalisedWork
+    hierarchy: WorkHierarchy
+    concepts: list[WorkConcept]
+
+
+class GraphWorksExtractor(GraphBaseExtractor):
+    def get_es_works(self, work_ids: list[str]) -> dict:
+        es_client = get_client("graph_extractor", "2025-05-01", True)
+        index_name = get_standard_index_name(
+            config.ES_DENORMALISED_INDEX_NAME, "2025-05-01"
```

Comment:
Can 2025-05-01 be derived from the environment? Or at least, as a member of the class, rather than hard-coded twice in this function